static/app/views/organizationContextContainer.tsx

```diff
@@ -16,329 +14,213 @@ import {ORGANIZATION_FETCH_ERROR_TYPES} from 'sentry/constants';
-      // retrieve initial state from store
-      return {
-        ...OrganizationStore.get(),
-        prevProps,
-      };
-    }
-
-    return {
-      loading: true,
-      error: null,
-      errorType: null,
-      organization: null,
-      prevProps,
-    };
-  }
-
+/**
+ * There are still a number of places where we consume the lgacy organization
+ * context. So for now we still need a component that provides this.
+ */
+class LegacyOrganizationContextProvider extends Component<{value: Organization | null}> {
   static childContextTypes = {
     organization: SentryTypes.Organization,
   };
 
-  constructor(props: Props) {
-    super(props);
-    this.state = OrganizationContextContainer.getDefaultState(props);
-  }
-
   getChildContext() {
-    return {
-      organization: this.state.organization,
-    };
-  }
-
-  componentDidMount() {
-    this.fetchData(true);
+    return {organization: this.props.value};
   }
 
-  componentDidUpdate(prevProps: Props) {
-    const remountPrevProps: State['prevProps'] = {
-      orgId: prevProps.params.orgId,
-      organizationsLoading: prevProps.organizationsLoading,
-      location: prevProps.location,
-    };
-
-    if (OrganizationContextContainer.shouldRemount(remountPrevProps, this.props)) {
-      this.remountComponent();
-    }
-  }
-
-  componentWillUnmount() {
-    this.unlisteners.forEach(callIfFunction);
-  }
-
-  unlisteners = [
-    OrganizationStore.listen(data => this.loadOrganization(data), undefined),
-  ];
-
-  remountComponent = () => {
-    this.setState(
-      OrganizationContextContainer.getDefaultState(this.props),
-      this.fetchData
-    );
-  };
-
-  isLoading() {
-    // In the absence of an organization slug, the loading state should be
-    // derived from this.props.organizationsLoading from OrganizationsStore
-    if (!getOrganizationSlug(this.props)) {
-      return this.props.organizationsLoading;
-    }
-
-    return this.state.loading;
+  render() {
+    return this.props.children;
   }
+}
 
-  fetchData(isInitialFetch = false) {
-    const orgSlug = getOrganizationSlug(this.props);
-
+/**
+ * Responsible for a number of organization related things:
+ *
+ * - Will load the current organization into the OrganizationStore if not
+ *   already present. This will often already be loaded by the
+ *   `preload-data.html` template
+ *
+ * -
+ */
+function OrganizationContextContainer({
+  includeSidebar,
+  useLastOrganization,
+  children,
+}: OrganizationContextContainerProps) {
+  const api = useApi();
+  const configStore = useLegacyStore(ConfigStore);
+
+  const {organizations} = useLegacyStore(OrganizationsStore);
+  const {organization, loading, error, errorType} = useLegacyStore(OrganizationStore);
+
+  const hasMadeFirstFetch = useRef(false);
+
+  const lastOrganization = useLastOrganization
+    ? configStore.lastOrganization ?? organizations[0]?.slug
+    : null;
+
+  const routes = useRoutes();
+  const params = useParams<{orgId?: string}>();
```

Comment:
Dont we always expect orgId to be present here? If we need undefined I would type the value as 
```suggestion
  const params = useParams<{orgId: string|undefined}>();
```